editoast/src/views/rolling_stock.rs

```diff
@@ -370,21 +371,32 @@ async fn update(
         .await?;
     assert_rolling_stock_unlocked(&previous_rolling_stock)?;
 
-    let mut new_rolling_stock = Into::<Changeset<RollingStockModel>>::into(rolling_stock_form)
-        .update(&mut db_pool.get().await?, rolling_stock_id)
-        .await
-        .map_err(|e| map_diesel_error(e, name.clone()))?
-        .ok_or(RollingStockError::KeyNotFound {
-            rolling_stock_key: RollingStockKey::Id(rolling_stock_id),
-        })?;
-
-    if new_rolling_stock != previous_rolling_stock {
-        new_rolling_stock.version += 1;
-        new_rolling_stock
-            .save(&mut db_pool.get().await?)
-            .await
-            .map_err(|err| map_diesel_error(err, name))?;
-    }
+    let new_rolling_stock = db_pool
+        .get()
+        .await?
+        .transaction::<_, InternalError, _>(|conn| {
+            async move {
+                let mut new_rolling_stock =
+                    Into::<Changeset<RollingStockModel>>::into(rolling_stock_form)
+                        .update(&mut conn.clone(), rolling_stock_id)
+                        .await
+                        .map_err(|e| map_diesel_error(e, name.clone()))?
+                        .ok_or(RollingStockError::KeyNotFound {
+                            rolling_stock_key: RollingStockKey::Id(rolling_stock_id),
+                        })?;
+
+                if new_rolling_stock != previous_rolling_stock {
```

Comment:
Shouldn't we grab `previous_rolling_stock` from inside the transaction as well? Otherwise we might be working with a stale `previous_rolling_stock`.

Reply:
Indeed, I believe you're right. Thanks for catching that. Done in [dc3b78b](https://github.com/OpenRailAssociation/osrd/pull/9207/commits/dc3b78b710a2023514c7fdf0c26a7214c0285f86).